core/src/loader.rs

```diff
@@ -1019,7 +1035,43 @@ impl<'gc> Loader<'gc> {
 
                     let movie = SwfMovie::from_data(&body, url.to_string(), loader_url)?;
                     player.lock().unwrap().mutate_with_update_context(|uc| {
+                        // Make a copy of the properties on the root, so we can put them back after replacing it
+                        let mut root_properties: IndexMap<AvmString, Value> = IndexMap::new();
+                        if let Some(root) = uc.stage.root_clip() {
+                            let root_val = root.object();
+                            if let Value::Object(root_object) = root_val {
+                                let mut activation = Activation::from_nothing(
+                                    uc.reborrow(),
+                                    ActivationIdentifier::root("unknown"),
+                                    root,
+                                );
+                                for key in root_object.get_keys(&mut activation, true) {
```

Comment:
Do we know if it should include hidden keys?
Can probably test it using `ASSetPropFlags`

Reply:
Yes, Kong's API makes all the properties it sets hidden. I'm still not certain why though.

Reply:
For more context

```actionscript
   static function onAbstractLoadStart(target)
   {
      target._lockroot = true;
      for(var i in _global.kongregate_vars)
      {
         target[i] = _global.kongregate_vars[i];
         _global.ASSetPropFlags(target,i,7);
         _global[i] = _global.kongregate_vars[i];
         _global.ASSetPropFlags(_global,i,7);
      }
      com.kongregate.common.util.Log.init();
      com.kongregate.common.util.Log.info("Creating services...");
      target.kongregateServices = _global.kongregateServices = com.kongregate.client.KongregateServices.getInstance();
      target.kongregateScores = _global.kongregateScores = com.kongregate.client.HighScoreServices.getInstance();
      target.kongregateStats = _global.kongregateStats = com.kongregate.client.StatServices.getInstance();
      target.kongregateChat = _global.kongregateChat = com.kongregate.client.ChatServices.getInstance();
      target.kongregateMtx = _global.kongregateMtx = com.kongregate.client.MicrotransactionServices.getInstance();
      target.kongregate = _global.kongregate = {services:com.kongregate.client.KongregateServices.getInstance(),scores:com.kongregate.client.HighScoreServices.getInstance(),stats:com.kongregate.client.StatServices.getInstance(),chat:com.kongregate.client.ChatServices.getInstance(),mtx:com.kongregate.client.MicrotransactionServices.getInstance()};
      _global.ASSetPropFlags(_global,["kongregate","kongregateServices","kongregateScores","kongregateStats","kongregateChat","kongregateMtx"],7);
      _global.ASSetPropFlags(target,["kongregate","kongregateServices","kongregateScores","kongregateStats","kongregateChat","kongregateMtx"],7);
      AbstractAPIBootstrap._subclassListener.onLoadStart(target);
   }
   ```